streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -533,10 +571,16 @@
         " WARNING: This config is deprecated and will be removed in 3.0.0 release.";
 
 
-    private static final String[] NON_CONFIGURABLE_CONSUMER_DEFAULT_CONFIGS = new String[] {ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG};
-    private static final String[] NON_CONFIGURABLE_CONSUMER_EOS_CONFIGS = new String[] {ConsumerConfig.ISOLATION_LEVEL_CONFIG};
-    private static final String[] NON_CONFIGURABLE_PRODUCER_EOS_CONFIGS = new String[] {ProducerConfig.ENABLE_IDEMPOTENCE_CONFIG,
-                                                                                        ProducerConfig.MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION};
+    private static final String[] NON_CONFIGURABLE_CONSUMER_DEFAULT_CONFIGS =
+        new String[] {ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG};
+    private static final String[] NON_CONFIGURABLE_CONSUMER_EOS_CONFIGS =
+        new String[] {ConsumerConfig.ISOLATION_LEVEL_CONFIG};
+    private static final String[] NON_CONFIGURABLE_PRODUCER_EOS_CONFIGS =
+        new String[] {
+            ProducerConfig.ENABLE_IDEMPOTENCE_CONFIG,
+            ProducerConfig.MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION,
+            ProducerConfig.TRANSACTIONAL_ID_CONFIG
```

Comment:
This is a side improvement: add `transaction.id` as a non-configurable parameter as we overwrite it anyway